cylc/flow/parsec/empysupport.py

```diff
@@ -52,15 +53,29 @@ def empyprocess(
     os.chdir(dir_)
     ftempl = StringIO('\n'.join(flines))
     xtempl = StringIO()
-    interpreter = em.Interpreter(output=em.UncloseableFile(xtempl))
+    # Detect EmPy version by interpreter.file() args.
+    # TODO: Use importlib.metadata version() once we drop Python 3.7
+    if 'name' in inspect.signature(em.Interpreter.file).parameters:
+        # Empy 3
+        interpreter = em.Interpreter(
+            output=em.UncloseableFile(xtempl)
+        )
+    else:
+        # Empy 4
+        # dispatcher = False: raise errors to caller
+        interpreter = em.Interpreter(
+            output=em.UncloseableFile(xtempl),
+            dispatcher=False
+        )
```

Comment:
If agreed on enforcing EmPy version 4 then we can drop the back-compat bit here.